Source/DataSources/GeoJsonDataSource.js

```diff
@@ -400,6 +415,9 @@ define([
         polygon.outlineColor = outlineColorProperty;
         polygon.outlineWidth = widthProperty;
         polygon.material = material;
+        if (!options.terrain) {
+            polygon.height = 0;
```

Comment:
You should move this to an else block after the if statement on 429, you only want the height to be 0 if perPositionHeight is not set to true.